Tensile/KernelWriterAssembly.py

```diff
@@ -7673,10 +7913,10 @@ def localReadDo(self, kernel, bufferIdx, iui, epsi, uIdx, tP):
             comment = "L -> Reg lro=%d swapByteOffset=%u ti=%u vIdx=%u rIdx=%u oIdx=%u buffer=%u iui=%u"\
                 % (tP["localReadOffset"], tP["localReadSwapByteOffset"], kernel["SubGroup%u" % tP["tensorIdx"]], vIdx, rIdx, oIdx, bufferIdx, iui)
             
-            isHighBits = 0 if kIdx % 2 == 0 else 1
+            isHighBits = 0 if rIdx % 2 == 0 else 1
             localReadCode.addCode(Code.LocalReadInst(instruction.toCodeInst(paramTuple, 0, isHighBits), comment))
 
-            if kIdx % 2 == 1:
+            if rIdx % 2 == 1:
               # pack 2 half-dword into one
               if kernel["MatrixInstruction"] and \
                  kernel["AssertSummationElementMultiple"] % kernel["MatrixInstK"] != 0 and \
```

Comment:
arbitrary-k size support has been put in `mfmaIter()` in upstream develop so this if-conditional can be gone if you do a rebase